include/utils/str.h

```diff
@@ -119,30 +124,40 @@ void str_append_c(str *dst, const char *src);
  */
 void str_append_buf(str *dst, const char *buf, size_t len);
 
+/**
+ * @brief Replace the string object data with a copy of the C string (null terminated).
+ */
+void str_copy_c(str *dst, const char *src);
```

Comment:
I specifically dropped this, as it tends to make things weird by reusing variable for different things. str_from_c() is there instead to create a new string from C string. Is there some specific place where you need this ?